examples/waspleau/Notes.md

```diff
@@ -0,0 +1,198 @@
+
+```
+Server: /Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/node_modules/.prisma/client/index.js:3
+Server (stderr):     throw new Error(
+Server (stderr):           ^
+Server (stderr):
+Server (stderr): Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
+Server (stderr): In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
+Server (stderr):     at new PrismaClient (/Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/node_modules/.prisma/client/index.js:3:11)
+Server (stderr):     at createDbClient (file:///Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/src/dbClient.js:5:24)
+Server (stderr):     at file:///Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/src/dbClient.js:11:18
+Server (stderr):     at ModuleJob.run (internal/modules/esm/module_job.js:152:23)
+Server (stderr):     at async Loader.import (internal/modules/esm/loader.js:166:24)
+Server (stderr):     at async Object.loadESM (internal/process/esm_loader.js:68:5)
+Server (stderr): [nodemon] app crashed - waiting for file changes before starting...
+```
+
+Therefore, I assumed I needed to migrate, however...
+
+### Running `wasp db migrate-dev` without any entities defined
+
+```
+Database successfully set up!
+Copying migrations folder from Wasp to Prisma project...
+Done copying migrations folder.
+Performing migration...
+Prisma schema loaded from ../db/schema.prisma
+Datasource "db": SQLite database "dev.db" at "file:./dev.db"
+
+Already in sync, no schema change or pending migration was found.
+
+Running generate... (Use --skip-generate to skip the generators)
+Error:
+You don't have any models defined in your schema.prisma, so nothing will be generated.
+You can define a model like this:
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+
+More information in our documentation:
+https://pris.ly/d/prisma-schema
+
+Error: Migrate dev failed: Migrate (dev) failed with exit code: 1
+```
+
+The fix was to add a dummy migration just to get past the Prisma error.
+
+
+| Bug captured here: https://github.com/wasp-lang/wasp/issues/398 |
+| --------------------------------------------------------------- |
+
+## 2) Axios version pinning
+
+```
+wasp: Error: Dependency conflict for user npm dependency (axios, ^0.21.1): Version must be set to the exactly the same version as the one wasp is using: ^0.21.1
+CallStack (from HasCallStack):
+  error, called at src/Generator/WebAppGenerator.hs:73:31 in waspc-0.2.2.2-B3XOTY4be7eFFZyFqJsQfH:Generator.WebAppGenerator
+```
+
+## 3) Dockerfile keeps getting overwritten. Any way to add to it? Do we want to use Docker Compose to allow for additional services like Redis?
+
+## 4) Dockerfile python stale/heroku container does not support M1/aarch64 [`heroku container:push` did not work]
```

Comment:
This is container that you provided yourself, by customizing the Dockerfile?

Reply:
Correct. We may have an actual issue in the generated Dockerfile, where `python` in `RUN apk add --no-cache build-base libtool autoconf automake python` no longer works and we need it to be `python3`. However, I am not 100% sure if this is M1 related or general. Could be a good test for someone else to verify.

Either way, I needed to update the Dockerfile and then use docker commands instead of heroku (but that part is specifically because of M1).

Reply:
Uff ok, great job on succeeding! Same as above, not sure if we should create an issue now or leave it for later.

Reply:
Thanks! How about I go ahead and create a [Waspleau example] issue and just have a list of some of these maybe things that we can either break out in the future into separate issues or ignore, mainly for tacking sake. Sound ok?